### ISSUE TITLE
To Do

### ISSUE THREAD

**Issue by develra** [sikuli/viz#10]
Michael Skirpan

Ian

Michael Aaron
- [x] Create main container for viz
- [x] Create a nav bar

Peyman

To-Do:

Phase 1 - on static data source
- [x] Make a left aside for fields
- [x] Make a right aside for templates
- [ ] Place a D3 React component into the main template
- [ ] Add react component for changing colors and scales
- [ ] Add react component for choosing binding data

**Comment by wannabeCitizen** [comment]
Today Peyman and I did a bunch of front end work.  We got the asides working and they are now collapsible and expandable.  However, we did run into 2 things we should tell you guys about:
1. The React components you were rendering in index.jsx were actually throwing errors.  Someone wrote in `document.getElementByClass()` and this is not correct since this is not a valid function and, even in its valid form of `document.getElementsByClassName()`, it returns a list, which React does not like.  Instead we should be using Id's here.  Also, it should be noted, that in some weird way, it turns out we don't even need to actually render some of these components in index.jsx because instead they are rendering themselves in their component .jsx file.  We stumbled upon this by trying to eliminate all the components to work on the interface from scratch, and they just wouldn't go away.
2. @ianks: We had some trouble with bower/gulp getting new dependencies into our scripts.  We used bower to install and save jquery-ui (which we deleted in the end anyway), but we ended up having to manually add the script to our `/boilerplate/scripts.jade` because gulp's build process wasn't adding it.  How can we automate this in the future?  I'm sure there is a way, but we spent 15 mins, couldn't get it, and decided to do it by hand for now.

Anyway, the front-end wire frame is coming along well.  Need to inject a d3 component to ensure we have that pattern down, then we can start building templates and adding the data backend.
